src/marine/oc_l2_radiance2ioda.py

```diff
@@ -0,0 +1,248 @@
+#!/usr/bin/env python3
+
+#
+# (C) Copyright 2023 UCAR
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+#
+
+import sys
+import argparse
+import netCDF4 as nc
+from datetime import datetime, timedelta
+import dateutil.parser
+import numpy as np
+from multiprocessing import Pool
+import os
+from pathlib import Path
+
+import pyiodaconv.ioda_conv_engines as iconv
+from pyiodaconv.orddicts import DefaultOrderedDict
+
+os.environ["TZ"] = "UTC"
+
+output_var_names = ["radiance"]
+
+locationKeyList = [
+    ("latitude", "float"),
+    ("longitude", "float"),
+    ("dateTime", "long")
+]
+
+GlobalAttrs = {
+    'odb_version': 1,
+    'converter': os.path.basename(__file__)
+}
+
+VarAttrs = DefaultOrderedDict(lambda: DefaultOrderedDict(dict))
+
+DimDict = {}
+
+VarDims = {}
+
+chan_number = range(1, 11)  # we have 10 channels
+
+
+def read_input(input_args):
+    """
+    Reads/converts a single input file, performing optional thinning also.
+
+    Arguments:
+
+        input_args: A tuple of input filename and global_config
+            input_file: The name of file to read
+            global_config: structure for global arguments related to read
+
+    Returns:
+
+        A tuple of (obs_data, GlobalAttrs) needed by the IODA writer.
+    """
+    data_in = {}
+    input_file = input_args[0]
+    global_config = input_args[1]
+    ncd = nc.Dataset(input_file, 'r')
+
+    # since OC L2 files are quite empty, need a mask applied immediately
+    # to avoid using too much memory)
+    Rrs_412 = ncd.groups['geophysical_data'].variables['Rrs_412'][:].ravel()
+    mask = Rrs_412 >= -3000  # min valid value
+
+    # Determine the lat/lon grid.
+    lons = ncd.groups['navigation_data'].variables['longitude'][:].ravel()[mask]
+    lats = ncd.groups['navigation_data'].variables['latitude'][:].ravel()[mask]
+    pixels = ncd.groups['navigation_data'].variables['longitude'][:].shape[1]
+    lines = ncd.groups['navigation_data'].variables['longitude'][:].shape[0]
+
+    # get basetime and time as a difference in seconds
+    global basetime
+    sla = ncd.groups['scan_line_attributes']
+    basetime = datetime(sla.variables['year'][0], 1, 1) + \
+        timedelta(days=int(sla.variables['day'][0]-1),
+                  milliseconds=int(sla.variables['msec'][0]))
+    time = (np.repeat(sla.variables['msec'][:].ravel(),
+            pixels).ravel() - sla.variables['msec'][0])/1000.0
+    time = time[mask]
+
+    obs_vars = (
+        'Rrs_412',
```

Comment:
would VIIRS have the same wavelengths?
If not, it would be nice to not hardcode these variable names and instead pull the wavelengths from the netcdf variable `sensor_band_parameters/wavelength` that's in the input file

Reply:
the issue with the sensor_band_parameters/wavelength variable, is, there are 24 bands in MODIS, but for just 10 of them the Rrs is derived, but you are right about the hard coding things! I'll fix it.